### PR TITLE
meta/autoid : fix the issue that MaxUint64 and MaxInt64 autoID is incorrectly allocated

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve? 
mysql 5.7 & mysql 8.0
```
DROP TABLE IF EXISTS t1;
CREATE TABLE t1 (pk BIGINT UNSIGNED AUTO_INCREMENT, PRIMARY KEY (pk));
INSERT INTO t1 VALUES (NULL), (18446744073709551615-1), (NULL);

ERROR 1467 (HY000): Failed to read auto-increment value from storage engine
```

tidb master
```
DROP TABLE IF EXISTS t1;
CREATE TABLE t1 (pk BIGINT UNSIGNED AUTO_INCREMENT, PRIMARY KEY (pk));
INSERT INTO t1 VALUES (NULL), (18446744073709551615-1), (NULL);

Query OK, 3 rows affected (0.01 sec)
Records: 3  Duplicates: 0  Warnings: 0
```

### What is changed and how it works?
Check the allocated value whether is the MaxUint64 or MaxInt64 before alloc.

### Check List 

Tests 

 - mysql-test In CI
 - correlated mysql_tset PR [#889 ](https://github.com/pingcap/tidb-test/pull/889) 

Related changes

 - Need to cherry-pick to the release branch

Release note

 - fix the MaxUint64 and MaxInt64 autoID should not allocated.

### CODE REVIEW

**Comment by AilinKid** [comment]
> Please add some test cases
Ok, Done.

**Comment by AilinKid** [comment]
> Please add some test cases

Done.

**Comment by AilinKid** [comment]
Attention: this PR need mysql-test #889 merged first

File: meta/autoid/autoid_test.go [hunk 1]
```diff
@@ -133,6 +134,14 @@ func (*testSuite) TestT(c *C) {
 	id, err = alloc.Alloc(3)
 	c.Assert(err, IsNil)
 	c.Assert(id, Equals, int64(6544))
+
+	// Test the MaxInt64 is the upper bound of `alloc` function but not `rebase`.
+	err = alloc.Rebase(3, int64(9223372036854775806), true)
```
Comment (Deardrops) [inline meta/autoid/autoid_test.go]: don't hard code here, use `math.MaxInt64 - 1`.

File: meta/autoid/autoid_test.go [hunk 2]
```diff
@@ -133,6 +134,14 @@ func (*testSuite) TestT(c *C) {
...
+	c.Assert(err, IsNil)
+	_, err = alloc.Alloc(3)
+	c.Assert(alloc, NotNil)
+	err = alloc.Rebase(3, int64(9223372036854775807), true)
```
Comment (Deardrops) [inline meta/autoid/autoid_test.go]: ditto

File: meta/autoid/autoid_test.go (hunk 1 above)
Comment (AilinKid) [reply meta/autoid/autoid_test.go]: Nice! Done.

File: meta/autoid/autoid_test.go (hunk 2 above)
Comment (AilinKid) [reply meta/autoid/autoid_test.go]: Done, PTAL.

**Pull request merged**